src/Common/src/CoreLib/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2564,8 +2594,8 @@ private bool IsAllowedJapaneseTokenFollowedByNonSpaceLetter(string tokenString,
             bool isLetter = char.IsLetter(ch);
             if (isLetter)
             {
-                ch = Culture.TextInfo.ToLower(ch);
-                if (IsHebrewChar(ch) && TokenMask == TokenType.RegularTokenMask)
+                ch = this.Culture.TextInfo.ToLower(ch);
+                if (!GlobalizationMode.Invariant && IsHebrewChar(ch) && TokenMask == TokenType.RegularTokenMask)
```

Comment:
> IsHebrewChar(ch) [](start = 51, length = 17)

nit: I am not sure if we need to break this scenario. even with Invariant, we used to allow parse some Hebrew numbers even if we are not using Hebrew calendar.

Reply:
To give you a little bit of background here: we did this in Mono as part of the Web Assembly work and the objection was to reduce the linked size of our `mscorlib.dll` as much as possible, moving all the globalization code that's not needed for US English into invariant mode.

I wrote a little tool that dumped IL size on a per-namespace as well as per-type basis and also looked at a general list of all the types still present in the generated code.  With the objective of "every single kilobyte counts" I then went over that list to see what can be removed.

Reply:
Thanks for clarifying. I am talking more about the scenario that is used to work with the Invariant mode and now will be broken. I think if we are parsing some date string which include some Hebrew numbers, I believe this used to succeed even with the Invariant mode as we are parsing the Hebrew numbers regardless if we are using the Hebrew calendar or not. What I am trying to say here, we should be explicit saying we don't have to support such scenarios with the Invariant mode if we go with this change.